### PR TITLE
I added :wa, :qa, :waq shortcuts commands

### PR DESCRIPTION
**Pull request by caiocutrim**

### CODE REVIEW

File: lib/ex.coffee [hunk 1]
```diff
@@ -199,9 +217,9 @@ class Ex
   substitute: (range, args) ->
     args = args.trimLeft()
     delim = args[0]
-    if /[a-z1-9\\"|]/i.test(delim)
+    if /[a-z]/i.test(delim)
       throw new CommandError(
-        "Regular expressions can't be delimited by alphanumeric characters, '\\', '\"' or '|'")
+        "Regular expressions can't be delimited by letters")
```
Comment (caiocutrim) [inline lib/ex.coffee]: I did fork this on branch master, why this change is over here?

File: lib/ex.coffee [hunk 2]
```diff
@@ -169,17 +176,28 @@ class Ex
       trySave(-> saveAs(fullPath)).then(deferred.resolve)
 
     deferred.promise
+  # save all files function
+  saveAll: ->
+    atom.workspace.saveAll()
+  # save all then quit
+  saveAllThenQuit: -  >
+    atom.workspace.saveAll()
+    @quitAll()
```
Comment (caiocutrim) [inline lib/ex.coffee]: I've created this function will be referenced on shortcut command that I've added...

File: lib/ex.coffee [hunk 3]
```diff
@@ -169,17 +176,28 @@ class Ex
...
     @write(args...)
 
   wq: (args...) =>
     @write(args...).then => @quit()
+  # save all files :wa command
+  wa: =>
+    @saveAll()
+  # save all, then quit shortcut
+  waq: =>
+    @saveAllThenQuit()
```
Comment (caiocutrim) [inline lib/ex.coffee]: Now we have :qa, :waq, :wa shortcuts :+1:

File: lib/ex.coffee [hunk 4]
```diff
@@ -71,10 +71,17 @@ class Ex
...
     atom.workspace.getActivePane().destroyActiveItem()
-
+  # quit function is used on :qa command shortcut
+  quitAll: ->
+    atom.workspace.getPanes()[0].destroy()
+  # both functions is referenced over here
+  # when this commands has trigered
   q: => @quit()
+  # ...and here
+  qa: => @quitAll()
```
Comment (caiocutrim) [inline lib/ex.coffee]: :+1:

File: lib/ex.coffee [hunk 5]
```diff
@@ -71,10 +71,17 @@ class Ex
   @registerCommand: (name, func) =>
     @singleton()[name] = func
 
+  # quit function is used on :q command
   quit: ->
     atom.workspace.getActivePane().destroyActiveItem()
-
+  # quit function is used on :qa command shortcut
+  quitAll: ->
```
Comment (jazzpi) [inline lib/ex.coffee]: In vim, this command is named `:quitall`. Could you rename it to that please? (And to the same in line 84)

File: lib/ex.coffee [hunk 6]
```diff
@@ -169,17 +176,28 @@ class Ex
       trySave(-> saveAs(fullPath)).then(deferred.resolve)
 
     deferred.promise
+  # save all files function
+  saveAll: ->
```
Comment (jazzpi) [inline lib/ex.coffee]: This command is named `wall` in vim. If you add a function to the `Ex` class, a command with the function name will be added to `ex-mode`, so all functions in here need to be named in the same way they are named in vim.

File: lib/ex.coffee [hunk 7]
```diff
@@ -169,17 +176,28 @@ class Ex
...
+    atom.workspace.saveAll()
+  # save all then quit
+  saveAllThenQuit: ->
```
Comment (jazzpi) [inline lib/ex.coffee]: This command is called `wqall` or `xall`.

File: lib/ex.coffee [hunk 8]
```diff
@@ -169,17 +176,28 @@ class Ex
...
   w: (args...) =>
     @write(args...)
 
   wq: (args...) =>
     @write(args...).then => @quit()
+  # save all files :wa command
+  wa: =>
+    @saveAll()
+  # save all, then quit shortcut
+  waq: =>
```
Comment (jazzpi) [inline lib/ex.coffee]: This command is called `waq`.

File: lib/ex.coffee (hunk 1 above)
Comment (jazzpi) [inline lib/ex.coffee]: I have no clue :confused: Maybe just manually revert the change?

**Comment by jazzpi** [comment]
I added some comments on the diff again, but apart from the command names this is good work, thanks :smile:

If you could add specs too, that would be great, otherwise I'll add some after this is merged.

File: lib/ex.coffee (hunk 1 above)
Comment (caiocutrim) [inline lib/ex.coffee]: I don't know how...

File: lib/ex.coffee (hunk 5 above)
Comment (caiocutrim) [inline lib/ex.coffee]: okay :)

File: lib/ex.coffee (hunk 7 above)
Comment (caiocutrim) [inline lib/ex.coffee]: I will rename it :+1:

**Comment by caiocutrim** [comment]
Thank you too, I'm glad to help with something :)

File: lib/ex.coffee (hunk 1 above)
Comment (jazzpi) [inline lib/ex.coffee]: Copypaste the lines from master and replace them? I think that's your best shot now.
